lib/main/lang/projectService.ts

```diff
@@ -162,7 +163,13 @@ declare module "${modulePath}"{
         fs.writeFileSync(defLocation, joinedDtsCode);
     }
 
+    let tsFilesWithInvalidEmit = outputs
+        .filter((o) => o.emitError) // only select outputs that produced an emit error
+        .map((o) => o.errors.map((e) => e.filePath)) // extract file paths from errors
+        .reduce((a, b) => a.concat(b), []); // flatten the array
+
```

Comment:
I meant `filePath` here : https://github.com/TypeStrong/atom-typescript/blob/master/lib/main/lang/modules/building.ts#L52-L57 

So that an `emitOutput` is associated with the `sourceFileName` easily and we don't need to do this lookup again later :rose: